### ISSUE TITLE
Solidity 0.5 support

### ISSUE THREAD

**Issue by hardlydifficult** [unlock-protocol/unlock#1078]
**Describe the solution you'd like**
We would like to upgrade to the latest version of Solidity.  This is not critical, but a nice to have improvement.

**Additional context**
Changing from Solidity 0.4 to 0.5 includes breaking changes.  Many .sol files cannot be written in such a way as to support both versions (some can, like ZOS's `Initializable`).

At the moment there is one blocker preventing us from completing the upgrade: https://github.com/OpenZeppelin/openzeppelin-eth specifically the `Ownable` implementation (the other import, `ERC165`, is available in openzeppelin-solidity).

**Comment by nfurfaro** [comment]
@hardlydifficult Can you elaborate on the reason we're blocked? I know we discussed this briefly on our first call, but I don't remember what it was.

**Comment by hardlydifficult** [comment]
We use https://github.com/OpenZeppelin/openzeppelin-eth/blob/master/contracts/ownership/Ownable.sol which is still coded for Solidity 0.4.  It looks like the repository may not be maintained any longer.  Almost everything in openzeppelin-eth is also in openzeppelin-solidity, however for the `Ownable` implementation they used a different initialization approach (https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/ownership/Ownable.sol).  

We need an `Ownable` implementation which supports Solidity 0.5 and uses ZOS's `Initializable` (or similar).  Of course we could copy-paste and then make the necessary adjustments.

**Comment by hardlydifficult** [comment]
Once this PR is accepted we can pick this task up again: https://github.com/OpenZeppelin/openzeppelin-eth/pull/43

**Comment by nfurfaro** [comment]
@hardlydifficult This can probably proceed now with the release of oz-eth v2.1.2. See #1223

**Comment by hardlydifficult** [comment]
\o/
On it.

**Comment by hardlydifficult** [comment]
I'm blocked on ZOS `TestHelper`.

The plan for this issue is to upgrade Truffle but continue using Solidity 0.4.25.  Once that's complete we can then create a PR to upgrade Solidity to 0.5.3.  The Truffle upgrade appears to cause the ZOS `TestHelper` to fail with `Cannot read property 'bind' of undefined at allPromisesOrError (node_modules/zos/lib/utils/async.js:36:11)`.

This may be related to this bug: https://github.com/zeppelinos/zos/issues/593

Here is a branch with the Truffle update applied: https://github.com/hardlydifficult/unlock/tree/hardlydifficult-truffleUpdate
You can see the errors in the Travis build: https://travis-ci.com/hardlydifficult/unlock/builds/99008911
